tests/harness/go/main/harness.go

```diff
@@ -27,28 +27,45 @@ func main() {
 	_, isIgnored := da.Message.(*cases.MessageIgnored)
 
 	msg, hasValidate := da.Message.(interface {
-		Validate() error
+		Validate(bool) error
 	})
 
+	var multierr error
 	if isIgnored {
 		// confirm that ignored messages don't have a validate method
 		if hasValidate {
-			err = fmt.Errorf("ignored message has Validate() method")
+			err = fmt.Errorf("ignored message has Validate(bool) method")
 		}
 	} else if !hasValidate {
-		err = fmt.Errorf("non-ignored message is missing Validate()")
+		err = fmt.Errorf("non-ignored message is missing Validate(bool)")
 	} else {
-		err = msg.Validate()
+		err = msg.Validate(false)
+		multierr = msg.Validate(true)
 	}
-	checkValid(err)
+	checkValid(err, multierr)
 }
 
-func checkValid(err error) {
-	if err == nil {
+func checkValid(err, multierr error) {
+	if err == nil && multierr == nil {
 		resp(&harness.TestResult{Valid: true})
 	} else {
-		resp(&harness.TestResult{Reason: err.Error()})
+		resp(&harness.TestResult{Reason: err.Error(), AllReasons: mergeReasons(nil, multierr)})
+	}
+}
+
+func mergeReasons(reasons []string, err error) []string {
+	multi, ok := err.(interface{ AllErrors() []error })
+	if !ok {
```

Comment:
Shouldn't this be some kind of fatal error? If the provided multierr doesn't implement the `AllErrors()` method, something is broken.

Reply:
Good point, thanks; I'll modify this.